core/arch/x86/encode.c

```diff
@@ -2596,7 +2634,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     di.start_pc = cache_pc;
     di.final_pc = final_pc;
 
-    while (!encoding_possible(&di, instr, info)) {
+    while (!encoding_possible(&di, instr, info) || !encoding_meets_hints(instr, info)) {
```

Comment:
It seems better to put encoding_meets_hints inside encoding_possible, right?  For other callers.  What we're calling "hints" are really requirements.  As a user I would expect instr_is_encoding_possible to fail due to an un-met hint.

Reply:
Done.

Reply:
(Agree).